### ISSUE TITLE
Using DOCKER_HOST instead of docker socket.

### ISSUE THREAD

**Issue by sean-e-dietrich** [docksal/docksal#452]
Using CircleCI there is no access to the /var/run/docker.sock but you do have access to the `$DOCKER_HOST` would it be possible to check for that instead?

Specifically looking to run docksal with [Docker In Docker](https://hub.docker.com/_/docker/) Image 

Mainly to run `fin init` and then `fin behat` on the Circle CI test

**Comment by lmakarov** [comment]
If `DOCKER_HOST` is set, then both `docker` and `docker-compose` will use that instead of `/var/run/docker.sock`.

Also, see https://circleci.com/docs/2.0/building-docker-images/ for using Docker with CircleCI. I have not tried this myself, though.

**Comment by sean-e-dietrich** [comment]
so issue is that VHOST Proxy tries to mount `/var/run/docker.sock` and ends up returning that it's not available. I saw those instructions and tried to build similarly but running into an issue where it wants to mount the docker.sock socket.

Docksal can run though when the `machine` executor is being used though which is really nice! But not when the docker instance is running. CircleCI may eventually phase out the `machine` executor or end up charging more as they are trying to use more of the docker images.

**Comment by lmakarov** [comment]
In regards to `/var/run/docker.sock` and `docksal-vhost-proxy`. It does need direct access to Docker via the socket, since it listens to Docker events and there is no way to do that over an HTTP(S) endpoint.

I just realized, that `fin` (and thus any `docker` and `docker-compose` commands issued by `fin`) does not honor the `DOCKER_HOST` environment variable, and that is done on purpose. `fin` sets this variable internally based on the environment it operates in (VirtualBox/Linux/Native apps). We could allow overriding it via the global `$HOME/.docksal/docksal.env` file. This will make it possible to use `fin` to manage a different (remote) Docker environment. System services can then be installed there via `fin system reset` (used to be `fin reset system`).

Realistically, you don't really need vhost-proxy for a single build running on CircleCI. However, `fin` does expect system services to be installed and running (a recently introduced check).

To sum it up - using Docksal/`fin` in a container (CI runtime) is not something currently supported and will require some adjustments.

**Comment by achekulaev** [comment]
> Using CircleCI there is no access to the /var/run/docker.sock

I don't think that running `docksal-vhost-proxy` inside CircleCI build is a good idea. After all CircleCI is a continuous integration, not a server. Or am I missing some point?

**Comment by achekulaev** [comment]
I realized that I did not read the conversation thorough enough. So fin does not work on CirleCI right now, because it tries to start system containers?

**Comment by achekulaev** [comment]
What we could make is to make fin omit the check for system services if environment variable is set.
i.e. if `CI=true`

**Comment by achekulaev** [comment]
Got it. Ignoring `DOCKER_HOST` value was a conscious decision. Some users had invalid/old values for this variable stored somewhere atfer their experiments with Docker or Docker-based solutions, that led to mysterious issues the users were attributing to Docksal. Finding where exactly that environment variable was set can be cumbersome, so instead we decided to override it always. 

But I hear you. I guess what we should do in this case is to create a path to "override the override". I think about creating a variable `DOCKSAL_HOST`, that would actually be a `DOCKER_HOST` variable but will have 0 chance of being set by other software.

**Comment by lmakarov** [comment]
@achekulaev I came to the same conclusion and already had a stashed change for this. See my commit above.

**Comment by achekulaev** [comment]
Nice. Should we close this one then to be fixed in the upcoming release?
